database/build/list_dashboard.go

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package build
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+
+	api "github.com/go-vela/server/api/types"
+	"github.com/go-vela/types/constants"
+	"github.com/go-vela/types/database"
+	"github.com/go-vela/types/library"
+)
+
+// ListBuildsForDashboardRepo gets a list of builds by repo ID from the database.
+//
+//nolint:lll // ignore long line length due to variable names
+func (e *engine) ListBuildsForDashboardRepo(ctx context.Context, r *api.Repo, branches, events []string) ([]*library.Build, error) {
```

Comment:
is there an advantage to having this version of ListBuildsForRepo?

Reply:
yeah, the ListBuildsForRepo uses the [map filter](https://github.com/go-vela/server/blob/04f930869c93b4a6bef0f1233f9f59d081ec0b8b/database/build/list_repo.go#L49) Gorm function, which to my knowledge doesn't cover the `IN` keyword, but I could be wrong. Gorm docs are not the most intuitive. Second, I didn't want to have the created constraints baked in... as well as the pagination

Reply:
tangential: `lll` has been [turned off](https://github.com/go-vela/server/blob/04f930869c93b4a6bef0f1233f9f59d081ec0b8b/.golangci.yml#L137), so you shouldn't need that `//nolint` directive

Reply:
ah okay, cool.